### PR TITLE
feat(coingecko): add function for coingecko's `/coins/{id}/market_chart/range` endpoint

### LINKED ISSUE #26: Add a function for /coins/{id}/market_chart/range coingecko endpoint
For more details look at #4

### PR DESCRIPTION
**Pull request by codeproline**
Hey there,

This PR adds function for coingecko's `/coins/{id}/market_chart/range` endpoint.

Closes #26

### CODE REVIEW

File: packages/coingecko/src/query/utils.ts [hunk 1]
```diff
@@ -27,3 +29,33 @@ export function getNullableStringProperty(json: JSON.Obj, prop: string): string
 
   return (json.get(prop) as JSON.Value).toString();
 }
+
+export function normalizeTimestampPricePairArray(array: JSON.Arr): TimestampPricePair[] {
+  return array._arr.map<TimestampPricePair>((item) => {
```
Comment (Niraj-Kamdar) [inline packages/coingecko/src/query/utils.ts:34]: Nice to have this utils!

File: packages/coingecko/src/__tests__/e2e/integration.spec.ts [hunk 2]
```diff
@@ -79,4 +78,34 @@ describe("Coingecko", () => {
...
+    expect(result.errors).toBeFalsy();
+    expect(result.data).toBeTruthy();
+
+    // check presence of properties
+    expect(result.data?.coinMarketChartRange).toHaveProperty("prices");
+    expect(result.data?.coinMarketChartRange).toHaveProperty("market_caps");
+    expect(result.data?.coinMarketChartRange).toHaveProperty("total_volumes");
+
+    // check one item for each prop
+    expect(result.data?.coinMarketChartRange.prices[0]).toBeTruthy();
```
Comment (Niraj-Kamdar) [inline packages/coingecko/src/__tests__/e2e/integration.spec.ts:107]: Let's also add test for checking length of these arrays.
```
    expect(result.data?.coinMarketChartRange.prices.length).toBe(4);
```

**Review by Niraj-Kamdar — CHANGES_REQUESTED** [review]
LGTM! Let's just add a small test on

File: packages/coingecko/src/__tests__/e2e/integration.spec.ts [hunk 3]
```diff
@@ -79,4 +78,39 @@ describe("Coingecko", () => {
...
+    expect(result.data?.coinMarketChartRange.market_caps[0]).toBeTruthy();
+    expect(result.data?.coinMarketChartRange.total_volumes[0]).toBeTruthy();
+
+    // check length of result arrays
+    expect(result.data?.coinMarketChartRange.prices.length).toBe(4);
```
Comment (Niraj-Kamdar) [inline packages/coingecko/src/__tests__/e2e/integration.spec.ts:112]: Length check test should be done before we check the elements of the array

**Review by Niraj-Kamdar — CHANGES_REQUESTED** [review]

File: packages/coingecko/src/__tests__/e2e/integration.spec.ts (hunk 2 above)
Comment (codeproline) [reply packages/coingecko/src/__tests__/e2e/integration.spec.ts:107]: Thanks for your feedback, I've added the test cases.

**Review by Niraj-Kamdar — APPROVED** [review]
LGTM! Thanks for this!

**Comment by rihp** [comment]
500$ as reward for completing this issue have been sent out
0xc91d113b794a4e57c1d942905ba65ba2bcf878b1dd7ce65862612500b9d27115

**Pull request merged**